crates/chain/src/local_chain.rs

```diff
@@ -13,7 +13,20 @@ use bitcoin::BlockHash;
 ///
 /// The key represents the block height, and the value either represents added a new [`CheckPoint`]
 /// (if [`Some`]), or removing a [`CheckPoint`] (if [`None`]).
-pub type ChangeSet = BTreeMap<u32, Option<BlockHash>>;
+pub struct ChangeSet<B>(BTreeMap<u32, Option<B>>);
```

Comment:
```suggestion
pub struct ChangeSet<B = BlockId>(BTreeMap<u32, Option<B>>);
```

Do this with all the other public types. I.e. `LocalChain`, `CheckPoint`. So when the caller does not specify `B`, it defaults to `BlockId` (the original behavior). This reduces the number of files needed to be changed.